falcon/app.py

```diff
@@ -461,10 +462,28 @@ def add_middleware(self, middleware: object) -> None:
         #   the chance that middleware may be None.
         if middleware:
             try:
-                self._unprepared_middleware += middleware
+                middleware = list(middleware)  # type: ignore
             except TypeError:
                 # middleware is not iterable; assume it is just one bare component
-                self._unprepared_middleware.append(middleware)
+                middleware = [middleware]
+
+            if (
+                self._cors_enable
+                and len(
```

Comment:
any reason why not using any? I think it's clearer
```py
any(isinstance(mc, CORSMiddleware) for mc in self._unprepared_middleware + middleware)
```

Reply:
No, a simple `any` probably doesn't work like that because I'm looking for a `len` of more than 1, not nonzero.

Reply:
right. The cors middleware is not added here but in the constructor.

sorry looked only at this section. I guess there is no better option for it